packages/ra-core/src/form/FormWithRedirect.spec.tsx

```diff
@@ -40,42 +41,18 @@ describe('FormWithRedirect', () => {
             />
         );
 
-        expect(renderProp.mock.calls[3][0].pristine).toEqual(true);
-        expect(getByDisplayValue('Foo')).not.toBeNull();
-        // 4 times because the first initialization with an empty value
-        // triggers a change on the input which has a defaultValue
-        // This is expected and identical to what FinalForm does (https://final-form.org/docs/final-form/types/FieldConfig#defaultvalue)
-        expect(renderProp).toHaveBeenCalledTimes(4);
-    });
-
-    it('Does not make the form dirty when initialized from a record with a missing field and this field has an initialValue', () => {
```

Comment:
Because there are fundamental differences between `final-form` and `react-hook-form`. `react-hook-form` only knows about `defaultValue` and will never set a field as dirty if it has not been changed by the user